src/components/ChatApp/MessageComposer.react.js

```diff
@@ -77,322 +75,275 @@ class MessageComposer extends Component {
     micColor: PropTypes.string,
     focus: PropTypes.bool,
     actions: PropTypes.object,
+    enterAsSend: PropTypes.bool,
+    micInput: PropTypes.bool,
+    messagesByID: PropTypes.object,
+    messages: PropTypes.array,
+    openSnackBar: PropTypes.func,
   };
 
   constructor(props) {
     super(props);
     this.state = {
       text: props.dream ? `dream ${props.dream}` : '',
-      start: false,
-      stop: false,
-      open: false,
-      result: '',
-      animate: false,
+      isListening: false,
+      isSpeechDialogOpen: false,
+      speechToTextOutput: '',
       rows: 1,
-      recognizing: false,
       micAccess: false,
-      currentArrowIndex: -1, // current message indexed at -1, latest message at 0
+      currentMessageIndex: -1,
+      speechRecognitionTextcolor: '#000',
     };
-    this.flag = 1;
+    this.userMessageHistory = [];
   }
 
-  componentDidMount() {
-    const { threadID, speechOutputAlways } = this.props;
-    const micInputSetting = UserPreferencesStore.getMicInput();
-    if (micInputSetting) {
-      // Getting the Speech Recognition to test whether possible
+  componentWillMount() {
+    const { micInput } = this.props;
+    if (micInput) {
       const SpeechRecognition =
         window.SpeechRecognition ||
         window.webkitSpeechRecognition ||
         window.mozSpeechRecognition ||
         window.msSpeechRecognition ||
         window.oSpeechRecognition;
-      // Setting buttons accordingly
-      if (SpeechRecognition != null) {
-        this.Button = <MicIcon />;
-        this.speechRecog = true;
-      } else {
-        this.Button = <SendIcon />;
-        this.speechRecog = false;
-      }
+      this.speechToTextAvailable = SpeechRecognition;
     } else {
-      this.Button = <SendIcon />;
-      this.speechRecog = false;
+      this.speechToTextAvailable = false;
     }
+  }
 
+  componentDidMount() {
+    const { speechOutputAlways, enterAsSend } = this.props;
+    this.userMessageHistory = [];
     let testSkill = $.urlParam('testExample');
```

Comment:
The $ dependency has been handled by https://github.com/fossasia/chat.susi.ai/pull/1854 submitted by @fragm3  so I have not removed it.